integrations/micrometer/src/main/java/io/helidon/integrations/micrometer/MicrometerCdiExtension.java

```diff
@@ -0,0 +1,254 @@
+
+import io.helidon.common.servicesupport.cdi.CdiExtensionBase;
+import io.helidon.common.servicesupport.cdi.LookupResult;
+
+import io.micrometer.core.annotation.Counted;
+import io.micrometer.core.annotation.Timed;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.LongTaskTimer;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+
+/**
+ * CDI extension for handling Micrometer artifacts.
+ */
+public class MicrometerCdiExtension extends CdiExtensionBase<
+        MicrometerCdiExtension.MicrometerAsyncResponseInfo,
+        MicrometerCdiExtension.MicrometerRestEndpointInfo,
+        MicrometerSupport,
+        MicrometerSupport.Builder> {
+
+    private static final Logger LOGGER = Logger.getLogger(MicrometerCdiExtension.class.getName());
+
+    private static final List<Class<? extends Annotation>> METRIC_ANNOTATIONS
+            = Arrays.asList(Counted.class, Timed.class);
+
+    private final MeterRegistry meterRegistry;
+
+    /**
+     * Creates new extension instance.
+     */
+    public MicrometerCdiExtension() {
+        super(LOGGER, Set.of(Counted.class, Timed.class), MeterProducer.class, config ->
+                        MeterRegistryProducer.getMicrometerSupport(),
+                "micrometer");
+        meterRegistry = MeterRegistryProducer.getMeterRegistry();
+    }
+
+    @Override
+    protected <E extends Member & AnnotatedElement>
+    void register(E element, Class<?> clazz, LookupResult<? extends Annotation> lookupResult) {
+        Annotation annotation = lookupResult.getAnnotation();
+
+        if (annotation instanceof Counted) {
+            Counter counter = MeterProducer.produceCounter(meterRegistry, (Counted) annotation);
+            LOGGER.log(Level.FINE, () -> "Registered counter " + counter.getId().toString());
+        } else if (annotation instanceof Timed) {
+            Timed timed = (Timed) annotation;
+            if (timed.longTask()) {
+                LongTaskTimer longTaskTimer = MeterProducer.produceLongTaskTimer(meterRegistry, timed);
+                LOGGER.log(Level.FINE, () -> "Registered long task timer " + longTaskTimer.getId()
+                        .toString());
+            } else {
+                Timer timer = MeterProducer.produceTimer(meterRegistry, timed);
+                LOGGER.log(Level.FINE, () -> "Registered timer " + timer.getId()
+                        .toString());
+            }
+        }
+    }
+
+    /**
+     * Initializes the extension prior to bean discovery.
+     *
+     * @param discovery bean discovery event
+     */
+    @Override
+    protected void before(@Observes BeforeBeanDiscovery discovery) {
+        LOGGER.log(Level.FINE, () -> "Before bean discovery " + discovery);
+        super.before(discovery);
+
+        // Initialize our implementation
+        MeterRegistryProducer.clear();
+
+        // Register beans manually
```

Comment:
Nit: terminology is important; there aren't any beans at this stage

Reply:
Agreed. Inherited cruft via copy/paste from the original metrics code, but I should've paid closer attention. Fixed.